client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -37,9 +44,16 @@ export const getGlobalStylesInfoForSite = ( siteId: number | null ): GlobalStyle
 };
 
 export function useSiteGlobalStylesStatus( siteId: number ): GlobalStylesStatus {
+	const [ , currentUserHasGlobalStylesInPersonal ] = useExperiment(
+		'calypso_global_styles_personal'
+	);
```

Comment:
We should avoid calling explat from the FE since we're now exposing the group as a BE service through an API. We might need to create another endpoint for when site = 0 or reuse the same one with siteId = 0.

Reply:
The BE would just use the equivalent of `useExperiment` because the site has not been created yet (which also requires a special endpoint because the current one is attached to an existing site). I lean towards not centralizing this in the BE because we are not duplicating code, it's simple enough to maintain here in Calypso, and it's code we eventually are going to delete.

Reply:
Wouldn't we be having the problem of giving us the assignment of the current user instead of the site owner as mentioned [here](p1688126418558299-slack-C04DZ8M0GHW)?

Reply:
The site ID is not 0, is null. When that happens, is because the site has not been created. And if the site has not been created, then the current user would be the owner of the new site.